receiver/elasticsearchreceiver/integration_test.go

```diff
@@ -57,6 +58,27 @@ var (
 	}
 )
 
+func SimpleAPITest(t *testing.T, baseURL string) {
+	client, err := newElasticsearchClient(componenttest.NewNopTelemetrySettings(), Config{
+		HTTPClientSettings: confighttp.HTTPClientSettings{Endpoint: baseURL},
+	}, componenttest.NewNopHost())
+	require.NoError(t, err)
+	ctx := context.Background()
+
+	_, err = client.Nodes(ctx, []string{})
+	require.NoError(t, err)
+	_, err = client.NodeStats(ctx, []string{})
+	require.NoError(t, err)
+	_, err = client.ClusterHealth(ctx)
+	require.NoError(t, err)
+	_, err = client.IndexStats(ctx, []string{})
+	require.NoError(t, err)
+	_, err = client.ClusterMetadata(ctx)
+	require.NoError(t, err)
+	_, err = client.ClusterStats(ctx, []string{"_all"})
+	require.NoError(t, err)
+}
```

Comment:
This is a good idea for ensuring that we are hitting all the endpoints. However, I am wondering why it is necessary to do it this way. Is there a way to configure the test so that it will make these calls during the normal data collection of the receiver?

Reply:
We _should_ be executing the call [here](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/36479b5c55a97d80b044e5f6101a6f6aabc0e20b/receiver/elasticsearchreceiver/scraper.go#L132-L146). I'm not sure why the configuration already used in the test is not doing this. Any ideas?

Reply:
let me dive into this

Reply:
@djaglowski it is supposed to fail here[https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/36479b5c55a97d80b044e5f6101a6f6aabc0e20b/receiver/elasticsearchreceiver/integration_test.go#L80-L86](url). But seems like it only wait for some/any metrics from the consumer without checking/verifying each of them. That is why it fails silently.

Reply:
This failure can only be caught during the integration test. For unit test, since we mock the ES client, this checking could be "fake"

Reply:
There is also a result check [here](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/receiver/elasticsearchreceiver/integration_test.go#L130-L133), but seems like it was never enabled("require.NoError()"). I tried added the check but there are quite some issues like the following

1. The content in expected json for 7.9.3 is wrong, there is no ".geoip_databases" index by default. It exist in  7.16.3 but not 7.9.3
2. It seems like the result metrics is not stable which might due to ES is in "starting..." state. After I enable the check it fails 2 out 5 runs.